### ISSUE TITLE
Cannot access forwarded ports remotely

### ISSUE THREAD

**Issue by mheon** [containers/libpod#1119]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**Description**

Ports forwarded to a container with Podman are accessible locally, but cannot be accessed by remote hosts.

**Steps to reproduce the issue:**

1. One machine 1: `podman run -d -p 80:80 -v <SOME_LOCAL_DIR>:/usr/share/nginx/html:ro nginx`

2. On machine 1: `iptables -t nat -S CNI-HOSTPORT-DNAT` to verify firewall rules are created

3. On machine 1: `telnet localhost 80` to verify local connectivity to HTTP server

4. On another machine: `telnet <IP_OF_MACHINE_1> 80` to verify remote connectivity to HTTP server

**Describe the results you received:**

IPTables output: 
`
-N CNI-HOSTPORT-DNAT
-A CNI-HOSTPORT-DNAT -p tcp -m comment --comment "dnat name: \"podman\" id: \"3bcb5133e8731ff31cc524756ca7b920bd7d69f0e5b5c675ee99ec0e316bc190\"" -m multiport --dports 80 -j CNI-DN-5d19ae988c93f23cd5102
-A CNI-HOSTPORT-DNAT -p udp -m comment --comment "dnat name: \"podman\" id: \"3bcb5133e8731ff31cc524756ca7b920bd7d69f0e5b5c675ee99ec0e316bc190\"" -m multiport --dports 80 -j CNI-DN-5d19ae988c93f23cd5102
`
Local connection to port 80 succeeds, remote connection fails

**Describe the results you expected:**

Local and remote connections both succeed

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

F28 packaged 0.7.1

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by mheon** [comment]
@dcbw PTAL

**Comment by vrothberg** [comment]
I made it work with a `registry:2` container running on port 5000, but only with `--network=host`. (openSUSE Leap 15, podman 0.7.2)

**Comment by mheon** [comment]
Talked with CNI maintainers - it seems like a problem with `firewalld` on Fedora. Not sure what openSUSE ships firewall-wise but it seems like a safe bet it's a conflict with firewall rules.

**Comment by vrothberg** [comment]
Huh, good to know. I will check again if the remote machine isn't on Leap 42.3 actually (no firewalld). Thanks, @mheon!

**Comment by fbruetting** [comment]
Any progress on this? I just tried out podman and wasted hours on searching for what I did wrong (I just started with containers) before seeing this here. 😒 `docker run -d -p 80:80 nginx` runs fine… Is using `-networking host` a usable workaround or how much unsafe is this really?

I’m on Fedora 28 Atomic Host, Podman v0.8.3 and containernetworking-plugins-0.7.3-1.

Is there a corresponding bug report at CNI?

**Comment by mheon** [comment]
#1431 might help, though I haven't had a chance to test it for this use case yet

**Comment by mheon** [comment]
Did a lot of testing with @baude today and #1431 seems to have resolved the issue. It's now merged and released in 0.9.1.1.

@fbruetting I'm going to close this as fixed in 0.9.1.1 - if you can reproduce after upgrading, feel free to reopen

**Comment by fbruetting** [comment]
Thank you very much, I’ll test! Do you maybe have an idea of how long it could take to be deployed in F28AH?

**Comment by mheon** [comment]
@fbruetting Hopefully we'll have a new build landed by the end of this week. If you don't want to wait, once the packages are built today/tomorrow, you should be able to grab them and add them as an rpm-ostree overlay.

**Comment by fbruetting** [comment]
`docker run -d -p 80:80 nginx` still doesn’t work on podman v0.9.1-3 from outside localhost, please reopen………

**Comment by mheon** [comment]
@fbruetting Can you open a new issue, and provide details on your environment there?

This shouldn't require `--net=host` so I'm suspecting it's something to do with iptables and/or firewalld

**Comment by fbruetting** [comment]
Tomorrow, (and firewalld isn’t active on my system).

**Comment by fbruetting** [comment]
Before starting a container (nothing else is active on the system):
```
$ sudo iptables -t nat -nvL

Chain PREROUTING (policy ACCEPT 895 packets, 45019 bytes)
 pkts bytes target     prot opt in     out     source               destination         
  896 45095 DOCKER     all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT 895 packets, 45019 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain OUTPUT (policy ACCEPT 192 packets, 14205 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT 192 packets, 14205 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 MASQUERADE  all  --  *      !docker0  172.17.0.0/16        0.0.0.0/0           

Chain DOCKER (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 RETURN     all  --  docker0 *       0.0.0.0/0            0.0.0.0/0
```

```
$ sudo iptables -nvL

Chain INPUT (policy ACCEPT 16584 packets, 93M bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain FORWARD (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
    0     0 DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
    0     0 ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0           

Chain OUTPUT (policy ACCEPT 9123 packets, 779K bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain DOCKER (1 references)
 pkts bytes target     prot opt in     out     source               destination         

Chain DOCKER-ISOLATION (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0
```


After starting `podman run -d -p 80:80 nginx`:
```
$ sudo iptables -t nat -nvL

Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 CNI-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL
  912 46280 DOCKER     all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 CNI-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL
    0     0 DOCKER     all  --  *      *       0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 CNI-HOSTPORT-MASQ  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* CNI portfwd requiring masquerade */
    0     0 MASQUERADE  all  --  *      !docker0  172.17.0.0/16        0.0.0.0/0           
    0     0 CNI-7f2861d14559a75555a20194  all  --  *      *       10.88.0.0/16         0.0.0.0/0            /* name: "podman" id: "8fbbd69b948383d7ca4a403d3d63ee5da13c61d3436561830b4b291f9693ad49" */

Chain CNI-7f2861d14559a75555a20194 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "8fbbd69b948383d7ca4a403d3d63ee5da13c61d3436561830b4b291f9693ad49" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "8fbbd69b948383d7ca4a403d3d63ee5da13c61d3436561830b4b291f9693ad49" */

Chain CNI-DN-7f2861d14559a75555a20 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 CNI-HOSTPORT-SETMARK  tcp  --  *      *       10.88.0.42           0.0.0.0/0            tcp dpt:80
    0     0 CNI-HOSTPORT-SETMARK  tcp  --  *      *       127.0.0.1            0.0.0.0/0            tcp dpt:80
    0     0 DNAT       tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:80 to:10.88.0.42:80

Chain CNI-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 CNI-DN-7f2861d14559a75555a20  tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            /* dnat name: "podman" id: "8fbbd69b948383d7ca4a403d3d63ee5da13c61d3436561830b4b291f9693ad49" */ multiport dports 80

Chain CNI-HOSTPORT-MASQ (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0            0.0.0.0/0            mark match 0x2000/0x2000

Chain CNI-HOSTPORT-SETMARK (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 MARK       all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* CNI portfwd masquerade mark */ MARK or 0x2000

Chain DOCKER (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 RETURN     all  --  docker0 *       0.0.0.0/0            0.0.0.0/0
```

```
$ sudo iptables -nvL

Chain INPUT (policy ACCEPT 47 packets, 3216 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain FORWARD (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      !10.88.0.42  10.88.0.42           0.0.0.0/0           
    0     0 DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
    0     0 DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
    0     0 ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0           

Chain OUTPUT (policy ACCEPT 38 packets, 10612 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain DOCKER (1 references)
 pkts bytes target     prot opt in     out     source               destination         

Chain DOCKER-ISOLATION (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0
```

**Comment by mheon** [comment]
@fbruetting Are you sure this is 0.9.1? It's still running manual IPtables commands, not using our new firewall code - `Running iptables command` no longer appears anywhere in our code...

If this is labelled 0.9.1, we might have a bad build running around somewhere.

**Comment by fbruetting** [comment]
Well…:
```
$ podman --version
podman version 0.9.1

$ rpm -q podman
podman-0.9.1-3.gitaba58d1.fc28.x86_64
```

I’m on most-recent Atomic Host:
```
$ rpm-ostree status

State: idle
AutomaticUpdates: disabled
Deployments:
● ostree://fedora-atomic:fedora/28/x86_64/atomic-host
                   Version: 28.20180917.0 (2018-09-17 02:21:22)
                BaseCommit: 17b11d74047ded1264a57555a64ae6fc5d02a332c556679bfcf32864fc91f11e
              GPGSignature: Valid signature by 128CF232A9371991C8A65695E08E7E629DB62FB1
           LayeredPackages: buildah exa fd-find git htop julia neovim powerline-go ripgrep zsh
```

**Comment by mheon** [comment]
Wait a second... I don't think the fix landed until 0.9.1.1

That would explain things.

**Comment by mheon** [comment]
I'll poke people tomorrow and see about getting things bumped to the newer version

**Comment by fbruetting** [comment]
So v0.9.1.1 > 0.9.1-3? O_o

**Comment by mheon** [comment]
`-3` is the RPM version of 0.9.1 - so it'll probably be `0.9.1.1-1` or similar

**Comment by fbruetting** [comment]
Ok, easy to misunderstand…
